app/controllers/api/storage.php

```diff
@@ -882,8 +888,15 @@
             );
         }
 
-        if (!empty($algorithm)) {
-            $source = $compressor->decompress($source);
+        switch ($algorithm) {
+            case 'zstd':
+                $compressor = new Zstd();
+                $source = $compressor->decompress($source);
+                break;
+            case 'gzip':
+                $compressor = new GZIP();
+                $source = $compressor->decompress($source);
+                break;
```

Comment:
Dont we need a default case?  
Because when we compress, we are defaulting to GZIP 
https://github.com/appwrite/appwrite/pull/3737/files#diff-02d3b2d220ab36c4733052fd1e3e017df738b47f7b3033dc8a93efbb7752436bR513